README
NAME
    CatalystX::SimpleLogin - Provide a simple Login controller which can be
    reused

SYNOPSIS
        package MyApp;
        use Moose;
        use namespace::autoclean;

        use Catalyst qw/
            +CatalystX::SimpleLogin
            Authentication
            Session
            Session::State::Cookie
            Session::Store::File
        /;
        extends 'Catalyst';

        __PACKAGE__->config(
            'Plugin::Authentication' => { # Auth config here }
        );

       __PACKAGE__->config(
            'Controller::Login' => { # SimpleLogin config here }
       );

       __PACKAGE__->setup;

ATTENTION!
    If you're new here, you should start by reading
    CatalystX::SimpleLogin::Manual, which provides a gentler introduction to
    using this code. Come back here when you're done there.

DESCRIPTION
    CatalystX::SimpleLogin is an application class Moose::Role which will
    inject a Catalyst::Controller which is an instance of
    CatalystX::SimpleLogin::Controller::Login into your application. This
    provides a simple login and logout page with the addition of only one
    line of code and one template to your application.

REQUIREMENTS
    A Catalyst application
    Working authentication configuration
    Working session configuration
    A view

CUSTOMISATION
    CatalystX::SimpleLogin is a prototype for CatalystX::Elements. As such,
    one of the goals is to make it easy for users to customise the provided
    component to the maximum degree possible, and also, to have a linear
    relationship between effort invested and level of customisation
    achieved.

    Three traits are shipped with SimpleLogin: WithRedirect, Logout, and
    RenderAsTTTemplate. These traits are set in the config:

       __PACKAGE__->config(
            'Controller::Login' => {
                traits => [qw/ Logout WithRedirect RenderAsTTTemplate /],
                login_form_args => { # see the login form },
       );

COMPONENTS
    *   CatalystX::SimpleLogin::Controller::Login - first point of call for
        customisation. Override the action configs to reconfigure the paths
        of the login or logout actions. Subclass to be able to apply method
        modifiers to run before / after the login or logout actions or
        override methods.

    *   CatalystX::SimpleLogin::TraitFor::Controller::Login::Logout -
        provides the "logout" action and associated methods. You can compose
        this manually yourself if you want just that action.

        This trait is set by default, but if you set another trait in your
        config, you will have to include it.

    *   CatalystX::SimpleLogin::TraitFor::Controller::Login::WithRedirect -
        provides the "login" action with a wrapper to redirect to a page
        which needs authentication, from which the user was previously
        redirected. Goes hand in hand with Catalyst::ActionRole::NeedsLogin

    *   CatalystX::SimpleLogin::TraitFor::Controller::Login::RenderAsTTTempl
        ate - sets the stash variable 'template' to point to a string
        reference containing the rendered template so that it's not
        necessary to have a login.tt template file.

    *   CatalystX::SimpleLogin::Form::Login - the HTML::FormHandler form for
        the login form.

    *   Catalyst::ActionRole::NeedsLogin - Used to cause a specific path to
        redirect to the login page if a user is not authenticated.

TODO
    Here's a list of what I think needs working on, in no particular order.

    Please feel free to add to or re-arrange this list :)

    Fix extension documentation
    Document all this stuff.
    Examples of use / customisation in documentation
    Fixing one uninitialized value warning in LoginRedirect
    Disable the use of NeedsLogin ActionRole when WithRedirect is not loaded

SOURCE CODE
        http://github.com/bobtfish/catalystx-simplelogin/tree/master

        git://github.com/bobtfish/catalystx-simplelogin.git

    Forks and patches are welcome. #formhandler or #catalyst (irc.perl.org)
    are both good places to ask about using or developing this code.

SEE ALSO
    *   Catalyst

    *   Moose and Moose::Role

    *   MooseX::MethodAttributes::Role - Actions composed from Moose::Role.

    *   CatalystX::InjectComponent - Injects the controller class

    *   HTML::FormHandler - Generates the login form

    *   Catalyst::Plugin::Authentication - Responsible for the actual heavy
        lifting of authenticating the user

    *   Catalyst::Plugin::Session

    *   Catalyst::Controller - Allows you to decorate actions with roles
        (E.g Catalyst::ActionRole::NeedsLogin)

    *   CatalystX::Component::Traits - Allows Moose::Role to be composed
        onto components from config

AUTHORS
    Tomas Doran (t0m) "bobtfish@bobtfish.net"
    Zbigniew Lukasiak
    Stephan Jauernick (stephan48) "stephan@stejau.de"
    Gerda Shank (gshank) "gshank@cpan.org"
    Florian Ragwitz "rafl@debian.org"
    Shlomi Fish
    Oleg Kostyuk (cub-uanic) "cub@cpan.org"

LICENSE
    Copyright 2009 Tomas Doran. Some rights reserved.

    This software is free software, and is licensed under the same terms as
    perl itself.



Changes
0.21 Tue, 22 December 2020 12:15:00 +0100
  * Change bugtracker to github.com

0.20 Wed, 14 June 2017 21:30:00 +0200
  * Fix spelling error (RT#116452, thanks Nick Morrott)
  * Don't use Moose::Autobox (fixes RT#113157)

0.19 Mon, 22 February 2016 18:16:00 +0100
  * Remove test dependency on Session::Store::File
  * Fix "remember" checkbox (Yaroslav Polyakov)
  * Made remember me behaviour extensible and configurable.
  * Rotate session id after login/logout
  * Catch exceptions thrown by the authenticate method

0.18 Sat, 21 July 2012 14:39:00 +0100
  * Stop depending on the now unused Catalyst::Controller::ActionRole
    RT#78500

0.17 Sun, 15 July 2012 18:23:00 +0100
  * Better fix for Catalyst versions >= 5.90013 RT#78340, one module
    and some documentation was missed.

0.16 Fri, 13 July 2012 15:41:00 +0100
  * Ensure that temp directories created in tests get removed.
  * Fix for Catalyst versions >= 5.90013 RT#78340

0.15 Tue, 6 Sept 2011 09:59:00 +0100
  * Various documentation fixes and improvements
  * Add tab index to forms
  * Fix login redirect to just use the URI, rather than trying to compose
    an action, as in some cases we may not have an action.
    (E.g. when you have a sub default : Private { action)

0.14 Tue, 12 Oct 2010 17:54:00 +0100
  * Large improvements in the documentation to pass all POD coverage tests
    and document examples well thanks to Shlomi Fish.

0.13 Thu, 07 Oct 2010 09:35:00 +0100
  * Fix the docs to make it clearer how to remove traits and use your own
    login form.

0.12
  * Fix test failures with various DBIC versions by relying on the DBIC
    dependency magic (RT#58307).

0.11
  * Add a chunk of documentation (Drew Taylour)

0.10  Wed, 02 Jul 2010 17:03:41 +0000
  * Add a clear_session_on_logout config setting which will blow away the
    contents of the session on logout.

0.09  Sat, 20 Feb 2010 18:05:05 +0000
  * Add actions for people using Chained controllers to chain off.
    /login/required and /login/not_required
  * Point new users to the manual at the top of the POD so they know it's there.
  * Add SQL::Translator as a test dependency.

0.08  Wed, 27 Jan 2010 22:02:11 +0000
  * Fix Login controller so that it works with the RequiresLogin action
    role without the WithRedirect trait composed.
  * Added redirect_after_login_uri configuration key to Controller::Login which
    can be used to set the path to redirect the user after login (if the
    WithRedirect trait is not used)
  * Added redirect_after_logout_uri configuration key to Controller::Login which
    can be used to set the path to redirect the user after logout.
  * Fix manual typo (Curtis 'Ovid' Poe)

0.07  Sun, 13 Dec 2009 20:00:23 +0000
  * Fix so that you can pass parameters to the login form from config to change
    the keys used in the $c->authenticate call, so that you can use an arbitrary
    DBIC schema result class.
  * Add tests with a DBIC using app.
  * Significant refactoring in the test suite to reduce code and template duplication
    between the test applications.

0.06  Fri, 11 Dec 2009 00:01:23 +0000
  * Fix Catalyst::ActionRole::NeedsLogin to correctly detach from
    action chains + tests.
  * Add an experimental controller for OpenID support.

0.05  Wed, 09 Dec 2009 22:50:23 +0000
  * WARNING: BREAKING CHANGE - The WithRedirect and RenderAsTTTemplate
    traits are now composed as default to reduce the amount of config
    needed in the tutorial, and as these are mostly what people want.
    If you don't want these traits, then you'll have to remove them with
    config.
  * Various documentation cleanups.
  * Remove dependency on Test::MockModule
  * Make OpenID support optional so that people don't get stuck installing
    Crypt::DH without the support modules which stop it taking forever.

0.04  Sun, 06 Dec 2009 13:25:23 +0000
  * Fix too low a dependency on HTML::Formhandler
  * Clean up a couple of things in the controller code to use $self
    rather than $c->controller('Login').

0.03  Sat, 14 Nov 2009 16:28:23 +0000
  * Fix dependency on Authentication::Credential::OpenID
  * Doc fixes

0.02  Sat, 07 Nov 2009 21:20:00 +0000
  * Much more documentation
  * Traits are automatically merged
  * Added OpenID support

0.01  Wed, 30 Sep 2009 02:17:00 +0200
  * Initial release.


Makefile.PL
use strict;
use warnings;
use inc::Module::Install 0.91;

use Module::Install::AuthorTests;
use Module::Install::AuthorRequires;

name 'CatalystX-SimpleLogin';
all_from 'lib/CatalystX/SimpleLogin.pm';

requires 'Moose';
requires 'Catalyst::Runtime' => '5.80013';
requires 'MooseX::MethodAttributes' => '0.18';
requires 'Catalyst::Action::REST' => '0.74';
requires 'Catalyst::Plugin::Authentication';
requires 'Catalyst::View::TT';
requires 'CatalystX::InjectComponent';
requires 'CatalystX::Component::Traits' => '0.13';
requires 'MooseX::Types::Common';
requires 'MooseX::Types';
requires 'MooseX::RelatedClassRoles' => '0.004';
requires 'HTML::FormHandler' => '0.28001';
requires 'namespace::autoclean';
requires 'Catalyst::Plugin::Session' => '0.35'; # Required as we use the 'Plugin::Session' config key in ::Manual
requires 'Try::Tiny' => '0.24';

test_requires 'Test::More'  => '0.94';
test_requires 'Class::Load' => '0.20';
test_requires 'Test::Exception';
test_requires 'File::Temp';
test_requires 'Catalyst::Action::RenderView';
test_requires 'Catalyst::Plugin::Session::State::Cookie';
test_requires 'HTTP::Request::Common';
test_requires 'Catalyst::ActionRole::ACL';
test_requires 'CatalystX::InjectComponent';
test_requires 'SQL::Translator';

author_requires 'Test::EOL' => '0.3';
author_requires 'Test::NoTabs';
author_requires 'Test::Pod' => '1.14';
author_requires 'Test::Pod::Coverage' => '1.08';
author_requires 'Catalyst::Model::DBIC::Schema';
author_requires 'Catalyst::Authentication::Store::DBIx::Class';

license 'perl';

resources bugtracker => 'https://github.com/bobtfish/catalystx-simplelogin/issues';
resources repository => 'https://github.com/bobtfish/catalystx-simplelogin';

tests 't/*.t';
author_tests 't/author';

if ($Module::Install::AUTHOR) {
    system("pod2text lib/CatalystX/SimpleLogin.pm > README") and die;
}

auto_install;
auto_provides;
WriteAll();

